### ISSUE TITLE
Wrong transitions in binaries

### ISSUE THREAD

**Issue by jozeffrtus** [sdispater/pytzdata#3]
pytzdata version >= `2017.1` has wrong timezone transitions data in compiled binary files. 
I found this with this pendulum usage, but since pendulum only uses pytzdata tz database, I report the issue here. The following code wrongly raises `pendulum.tz.exceptions.AmbiguousTime: The datetime 2018-06-20 19:50:00 is ambiguous.`.
```
import pendulum


pendulum.set_transition_rule(pendulum.TRANSITION_ERROR)
print(pendulum.from_timestamp(1529524200).replace(tzinfo='Asia/Dubai'))
```

The point is, that **Asia/Dubai** currently does not observe DST, but somehow, following 
`<Transition [2038-01-19 03:14:07 UTC, 2038-01-19 07:14:07 -> 2038-01-19 07:14:07]>`
transition is defined for the **Asia/Dubai** timezone according to timezone binary file. Observe that transition is weird in a way that no shift will be made at all (pre and post transition times are the same).

Such weird transitions (year=2038 with pre and post transition times being the same) occur also in **America/Lima**, **America/Guayaquil**, **America/Bogota** and maybe in many others.

Note that in pytzdata version `2016.10` (and pendulum `0.7.0`), everything seems to be ok (i.e., the code above does not raise error and computes correct date time), since there are no such weird 2038-year's transitions.

May it be the case, that binaries with timezones are wrongly compiled from the Olson tz database? 

Thanx, for handling this issue

Jozef

**Comment by jozeffrtus** [comment]
UPDATE:
Ok, now I realized, that this is related to [Year 2038 problem](https://en.wikipedia.org/wiki/Year_2038_problem)

**Comment by sdispater** [comment]
pytzdata only exposes what the Olson database provides and the information about this 2038 transition is present so there is not much I can do about it.

You can see that `pytz` also references this transition:

```python
>>> import pytz
>>> tz = pytz.timezone('Asia/Dubai')
>>> tz._utc_transition_times
[datetime.datetime(1, 1, 1, 0, 0),
 datetime.datetime(1919, 12, 31, 20, 18, 48),
 datetime.datetime(2038, 1, 19, 3, 14, 7)]
>>> tz._transition_info
[(datetime.timedelta(0, 13260), datetime.timedelta(0), 'LMT'),
 (datetime.timedelta(0, 14400), datetime.timedelta(0), '+04'),
 (datetime.timedelta(0, 14400), datetime.timedelta(0), '+04')]
```

So it's not specific to pytzdata or pendulum.

**Comment by jozeffrtus** [comment]
Thank you for the reply.
I understand that pytzdata only exposes Olson's database. But this 2038 transition is not present in Olson's database. This is the snippet from the [latest](https://www.iana.org/time-zones/repository/releases/tzdb-2017c.tar.lz) tz distribution, file `asia`:
```
# United Arab Emirates
# Zone	NAME		GMTOFF	RULES	FORMAT	[UNTIL]
Zone	Asia/Dubai	3:41:12 -	LMT	1920
			4:00	-	+04
Link Asia/Dubai Asia/Muscat	# Oman
```
Or am I missing something? Can you please provide me specific link to file and line number which defines this transition? 
For me, this issue seems rather as a bug during compiling Olson's plain text into binaries: either `zic` or its usage in `pytzdata` seems to be buggy.

**Comment by sdispater** [comment]
It appears that the transitions is added when compiling database files. It is related to the Year 2038 bug since the transition time is `2038-01-19 03:14:07 UTC` but I don't have much more insight regarding this.

But like I said, it is not specific to pyzdata since other libraries using the Olson database as a reference.

**Comment by jozeffrtus** [comment]
Ok, you are right that it is not specific to pytzdata - I found this commit for `zic.c` (utility which is used as timezone compiler for Olsons database)
https://github.com/eggert/tz/commit/9c32443c5d2a40542c637030323728abb6c350c4#diff-15e7bfdf51e1e4795ebcaed3d885e880
If understand it correctly, they tried to provide workaround for some bug in `Qt` library by inserting such 2038 transition into binary timezone compilation :-)

Could you possible provide some compensation for this? For example ignore this transition in `pendulum`/`pytzdata`?

**Comment by jozeffrtus** [comment]
Closing the issue as it is fixed in pendulum `1.4.1` release.
